BuildScript.Nuke/Components/ConfigurationData.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+// 
+
+using System;
+using System.Collections.Generic;
+
+namespace Remotion.BuildScript.Components;
+
+public class ConfigurationData
+{
+  public string NormalTestConfiguration { get; set; } = "";
+  public IReadOnlyCollection<string> SupportedTargetRuntimes { get; set; } = Array.Empty<string>();
+  public IReadOnlyCollection<string> SupportedExecutionRuntimes { get; set; } = Array.Empty<string>();
+  public IReadOnlyCollection<string> SupportedBrowsers { get; set; } = Array.Empty<string>();
+  public IReadOnlyCollection<string> SupportedDatabaseSystems { get; set; } = Array.Empty<string>();
+  public IReadOnlyCollection<string> SupportedPlatforms { get; set; } = Array.Empty<string>();
+  public IReadOnlyCollection<ProjectMetadata> ReleaseProjectFiles { get; set; } = Array.Empty<ProjectMetadata>();
+  public IReadOnlyCollection<TestProjectMetadata> UnitTestProjectFiles { get; set; } = Array.Empty<TestProjectMetadata>();
+  public IReadOnlyCollection<TestProjectMetadata> IntegrationTestProjectFiles { get; set; } = Array.Empty<TestProjectMetadata>();
+  public IReadOnlyCollection<TestProjectMetadata> TestProjectFiles { get; set; } = Array.Empty<TestProjectMetadata>();
+  public AssemblyMetadata AssemblyMetadata { get; set; } = null!;
+  public SemanticVersion SemanticVersion { get; set; } = null!;
```

Comment:
Imo those should be `init` properties, and the whole class should be instantiated and assigned to the property in the `ReadConfiguration` target instead of creating the instance in the property initializer and assigning the values in the target.

The big drawback would be that in theory the property containing the `ConfigurationData` instance must be nullable and thus checked on every usage, because the compiler can't follow the nuke target dependencies - thus it would be ok to leave it non-nullable and use `= null!;` for the property initializer.